microsoft/testsuites/power/common.py

```diff
@@ -32,69 +32,78 @@ def is_distro_supported(node: RemoteNode) -> None:
         )
 
 
-def verify_hibernation(node: RemoteNode, log: Logger) -> None:
+def verify_hibernation(
+    node: Node,
+    log: Logger,
+    ignore_call_trace: bool = False,
+) -> None:
+    hibernation_setup_tool = node.tools[HibernationSetup]
+    startstop = node.features[StartStop]
+
     node_nic = node.nics
     lower_nics_before_hibernation = node_nic.get_lower_nics()
     upper_nics_before_hibernation = node_nic.get_nic_names()
-    hibernation_setup_tool = node.tools[HibernationSetup]
     entry_before_hibernation = hibernation_setup_tool.check_entry()
     exit_before_hibernation = hibernation_setup_tool.check_exit()
     received_before_hibernation = hibernation_setup_tool.check_received()
     uevent_before_hibernation = hibernation_setup_tool.check_uevent()
-    startstop = node.features[StartStop]
+
+    # only set up hibernation setup tool for the first time
     hibernation_setup_tool.start()
-    startstop.stop(state=features.StopState.Hibernate)
+
+    try:
+        startstop.stop(state=features.StopState.Hibernate)
+    except Exception as ex:
+        try:
+            node.tools[Dmesg].get_output(force_run=True)
+        except Exception as e:
+            log.debug(f"error on get dmesg output: {e}")
+        raise LisaException(f"fail to hibernate: {ex}")
+
     is_ready = True
     timeout = 900
     timer = create_timer()
     while timeout > timer.elapsed(False):
-        is_ready, _ = wait_tcp_port_ready(
-            node.connection_info[constants.ENVIRONMENTS_NODES_REMOTE_ADDRESS],
-            node.connection_info[constants.ENVIRONMENTS_NODES_REMOTE_PORT],
-            log=log,
-            timeout=10,
-        )
-        if not is_ready:
+        if startstop.get_status() == AzureVMStatus.Deallocated.value:
+            is_ready = False
             break
     if is_ready:
-        raise LisaException("VM still can be accessed after hibernation")
+        raise LisaException("VM is not in deallocated status after hibernation")
+
     startstop.start()
+    dmesg = node.tools[Dmesg]
+    dmesg.check_kernel_errors(force_run=True, throw_error=not ignore_call_trace)
```

Comment:
The `ignore_call_trace` is used here only once. It can use the same name `throw_error`, or called `throw_error_on_kernel`. So it doesn't need `not`.